### PR TITLE
Feature model editing

### PR DESCRIPTION
**Pull request by sainsay**
Modelling and lights

## Description
added support for lights(spot, directional and point) both adding and removing.
added support for runtime loading of meshes.
added support for (basic) modelling 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/plugin/framegraph/frame_graph_manager.cpp [hunk 1]
```diff
@@ -40,14 +40,15 @@ namespace wmr
 
 		// Add required tasks to each frame graph
 		CreateDeferredPipeline();
-		CreateHybridRTPipeline();
-		CreateFullRTPipeline();
+		//CreateHybridRTPipeline();
```
Comment (tntmeijs) [inline src/plugin/framegraph/frame_graph_manager.cpp]: Please keep the pipeline enabled.

File: src/plugin/framegraph/frame_graph_manager.cpp [hunk 2]
```diff
@@ -40,14 +40,15 @@ namespace wmr
...
+		//CreateFullRTPipeline();
 
 		// Set-up the rendering pipelines (frame graph configuration)
-		for (auto& frame_graph : m_renderer_frame_graphs)
+		m_renderer_frame_graphs[0]->Setup( render_system );
```
Comment (tntmeijs) [inline src/plugin/framegraph/frame_graph_manager.cpp]: See previous comment.

File: src/plugin/parsers/light_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+#include <maya/MGlobal.h>
+#include <maya/MQuaternion.h>
+
+#include <DirectXMath.h>
+
+using namespace DirectX;
+
+namespace wr
+{
+	//template std::shared_ptr<LightNode> SceneGraph::CreateChild( std::shared_ptr<Node> const & parent, LightType tid, DirectX::XMVECTOR col );
```
Comment (tntmeijs) [inline src/plugin/parsers/light_parser.cpp]: BE GONE

File: src/plugin/parsers/light_parser.cpp [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+		
+		MFnLight fn_light( it->first );
+		MFnDagNode dagnode = fn_light.parent( 0, &status );
+		MObject object = dagnode.object();
+		MFnTransform transform_rhs( dagnode.object(), &status );
+		if( transform_rhs.object() != transform.object() )
+		{
+			return; // find_if returns last element even if it is not a positive result
+		}
+
```
Comment (tntmeijs) [inline src/plugin/parsers/light_parser.cpp]: Please have a look at your usage of unnecessary white spaces. It's just a bit inconsistent.

File: src/plugin/parsers/light_parser.hpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+#include <scene_graph/scene_graph.hpp>	
+#include <vector>
+#include <memory>
+
+
+namespace wr
+{
+	class LightNode;
+}
+
```
Comment (tntmeijs) [inline src/plugin/parsers/light_parser.hpp:13]: See previous comment regarding whitespace

File: src/plugin/parsers/light_parser.hpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	private:
+		//callbacks that require private access and are part of the ModelParser.
+		friend void AttributeLightTransformCallback( MNodeMessage::AttributeMessage msg, MPlug &plug, MPlug &otherPlug, void *clientData );
+
+		std::vector<std::pair<MObject, std::shared_ptr<wr::LightNode>>> m_object_transform_vector;
+		std::vector<std::pair<MObject, MCallbackId>> m_mesh_added_callback_vector;
+
+		Renderer& m_renderer;
+	};
+
```
Comment (tntmeijs) [inline src/plugin/parsers/light_parser.hpp]: See previous comment

File: src/plugin/renderer/material_manager.cpp [hunk 7]
```diff
@@ -34,10 +34,13 @@ void wmr::MaterialManager::Initialize()
 	wr::Material* internal_material = m_material_pool->GetMaterial( m_default_material_handle.m_id );
 	auto& texture_manager = renderer.GetTextureManager();
 
-	internal_material->SetAlbedo( texture_manager.GetDefaultTexture() );
-	internal_material->SetNormal( texture_manager.GetDefaultTexture() );
-	internal_material->SetMetallic( texture_manager.GetDefaultTexture() );
-	internal_material->SetRoughness( texture_manager.GetDefaultTexture() );
+	internal_material->SetUseConstantAlbedo( true );
+	internal_material->SetUseConstantMetallic( true );
+	internal_material->SetUseConstantRoughness( true );
```
Comment (tntmeijs) [inline src/plugin/renderer/material_manager.cpp:41]: No normal map will cause Hybrid to crash

**Review by tntmeijs — CHANGES_REQUESTED** [review]
Please review the comments I've added to the code.
It's minor stuff, but it does need to be changed...

File: src/plugin/framegraph/frame_graph_manager.cpp (hunk 1 above)
Comment (sainsay) [reply src/plugin/framegraph/frame_graph_manager.cpp]: enabled pipelines

File: src/plugin/parsers/light_parser.cpp (hunk 3 above)
Comment (sainsay) [reply src/plugin/parsers/light_parser.cpp]: removed commented code.

File: src/plugin/renderer/material_manager.cpp (hunk 7 above)
Comment (sainsay) [reply src/plugin/renderer/material_manager.cpp:41]: I set UseNormalTexture to false. this should work. otherwise, we add an issue for Wisp

**Review by tntmeijs — APPROVED** [review]
Thanks for making the changes. Looks good to me.

**Pull request merged**